File: ForgeSwift/ForgeCore/Sources/ForgeCore/Intelligence/AriaDayBrief.swift
import Foundation

// MARK: - AriaDayBrief
//
// One-line on-device "day brief" for Watch Home + complications. Pure and
// deterministic (SimRunner-friendly): no network, no LLM. Complements the
// mindfulness suggestion with a glanceable summary of readiness + sleep +
// the next best action.

public enum AriaDayBrief {

    /// Compose a single sentence the wrist can show under ARIA's greeting.
    public static func line(
        for context: WatchARIAContext,
        recommendation: MindfulnessRecommendation?
    ) -> String {
        let readinessBit: String
        if let score = context.readinessOverall, (context.readinessConfidence ?? 0) > 0 {
            let band = ReadinessBand(score: score)
            readinessBit = "Readiness \(score) (\(band.label))"
        } else {
            readinessBit = "Still gathering readiness"
        }

        let sleepBit: String
        if let quality = context.sleepQualityScore {
            if quality < 60 {
                sleepBit = "sleep ran short"
            } else if quality >= 80 {
                sleepBit = "sleep looked solid"
            } else {
                sleepBit = "sleep was okay"
            }
        } else if let minutes = context.sleepMinutes, minutes > 0 {
            let hours = minutes / 60.0
            sleepBit = String(format: "slept %.1fh", hours)
        } else {
            sleepBit = "sleep still syncing"
        }

        let actionBit: String
        if let rec = recommendation {
            actionBit = "Next: \(rec.durationLabel) \(rec.practice.displayName)"
        } else {
            actionBit = "Next: a short reset when you want it"
        }

        return "\(readinessBit) · \(sleepBit). \(actionBit)."
    }
}

File: backend/infra/lambda/aria_core/aria_day_brief.py
"""One-line day brief for Watch Home + complications.

Python port of ForgeCore's ``AriaDayBrief.swift``.
"""

from __future__ import annotations

from .mindfulness_suggestion import MindfulnessRecommendation
from .readiness_calculator import BAND_LABEL, band_for_score
from .watch_context import WatchContext


def line(context: WatchContext, recommendation: MindfulnessRecommendation | None) -> str:
    if context.readiness_overall is not None and (context.readiness_confidence or 0) > 0:
        band = band_for_score(context.readiness_overall)
        readiness_bit = f"Readiness {context.readiness_overall} ({BAND_LABEL[band]})"
    else:
        readiness_bit = "Still gathering readiness"

    if context.sleep_quality_score is not None:
        quality = context.sleep_quality_score
        if quality < 60:
            sleep_bit = "sleep ran short"
        elif quality >= 80:
            sleep_bit = "sleep looked solid"
        else:
            sleep_bit = "sleep was okay"
    elif context.sleep_minutes and context.sleep_minutes > 0:
        hours = context.sleep_minutes / 60.0
        sleep_bit = f"slept {hours:.1f}h"
    else:
        sleep_bit = "sleep still syncing"

    if recommendation is not None:
        action_bit = f"Next: {recommendation.duration_label} {recommendation.display_name}"
    else:
        action_bit = "Next: a short reset when you want it"

    return f"{readiness_bit} · {sleep_bit}. {action_bit}."
